java/unittests/infodynamics/measures/continuous/gaussian/OInfoCalculatorGaussianTester.java

```diff
@@ -0,0 +1,150 @@
+                                     {0.5,   1,   0,   0},
+                                     {  0,   0,   1, 0.5},
+                                     {  0,   0, 0.5,   1}};
+
+    OInfoCalculatorGaussian oCalc = new OInfoCalculatorGaussian();
+    oCalc.initialise(4);
+    oCalc.setCovariance(cov);
+    double oinfo = oCalc.computeAverageLocalOfObservations();
+
+    assertEquals(oinfo, 0, 1e-6);
+  }
+
+  /**
+   * Compare against the direct calculation of O-info as a sum of entropies using the
+   * entropy calculator.
+   */
+  public void testCompareWithEntropy() throws Exception {
+    double[][] cov = new double[][] {{1, 0.4, 0.3}, {0.4, 1, 0.2}, {0.3, 0.2, 1}};
+
+    OInfoCalculatorGaussian oCalc = new OInfoCalculatorGaussian();
+    oCalc.initialise(3);
+    oCalc.setCovariance(cov);
+    double oinfo = oCalc.computeAverageLocalOfObservations();
+
+    // Calculate using an entropy calculator and picking submatrices manually
+    EntropyCalculatorMultiVariateGaussian hCalc = new EntropyCalculatorMultiVariateGaussian();
+    hCalc.initialise(3);
+    hCalc.setCovariance(cov);
+    double oinfo_hCalc = hCalc.computeAverageLocalOfObservations();
+
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {0,1}, new int[] {0,1}));
+    oinfo_hCalc -= hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {0,2}, new int[] {0,2}));
+    oinfo_hCalc -= hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {1,2}, new int[] {1,2}));
+    oinfo_hCalc -= hCalc.computeAverageLocalOfObservations();
+
+    hCalc.initialise(1);
+    hCalc.setCovariance(new double[][] {{cov[0][0]}});
+    oinfo_hCalc += hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(1);
+    hCalc.setCovariance(new double[][] {{cov[1][1]}});
+    oinfo_hCalc += hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(1);
+    hCalc.setCovariance(new double[][] {{cov[2][2]}});
+    oinfo_hCalc += hCalc.computeAverageLocalOfObservations();
+
+    assertEquals(oinfo, oinfo_hCalc, 1e-6);
+
+  }
+
+  /**
+   * Confirm that the local values average correctly back to the average value
+   */
+  public void testLocalsAverageCorrectly() throws Exception {
+
+    int dimensions = 4;
+    int timeSteps = 1000;
+    OInfoCalculatorGaussian oCalc = new OInfoCalculatorGaussian();
+    oCalc.initialise(dimensions);
+
+    // generate some random data
+    RandomGenerator rg = new RandomGenerator();
+    double[][] data = rg.generateNormalData(timeSteps, dimensions,
+        0, 1);
+
+    oCalc.setObservations(data);
+
+    double oinfo = oCalc.computeAverageLocalOfObservations();
+    double[] oLocal = oCalc.computeLocalOfPreviousObservations();
+
+    System.out.printf("Average was %.5f\n", oinfo);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.